libevmcore/Assembly.h

```diff
@@ -24,6 +24,7 @@
 #include <iostream>
 #include <sstream>
 #include <libdevcore/Common.h>
+#include <libdevcore/SourceLocation.h>
```

Comment:
If you put it in devcore, it basically means "everyone needs it", which is certainly not true.
Since mix depends on solidity and solidity depends on evmcore, this is the lowest point you can put it. Furthermore, the current use-case for mix is to use SourceLocations only in connection with instructions and bytecode, whose realm is evmcore.
Note that evmcore might be wrongly named: It is not concerned with the EVM itself but rather with its bytecode.